### DISCUSSION TITLE
what about an element like ui.monitor()

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3201] [Ideas / Feature Requests]
`ui.monitor(value1,dict1,list1).on_valuechange(do something)`
Introducing a monitoring component to conveniently observe data variables would make it easy to monitor the changes in various variables within the program.

maybe  we could Introduce a reactive programming mode to this ui.monitor() like  ex4nicegui

**Comment by frankvp11** [comment]
I like the idea, but that's basically what the bindings already do. They append a bunch of elements to a list, and check if they've been updated(if I remember correctly). How would that be different from this?

**Comment by v479038280** [comment]
It's similar to binding,but binding has to be used with some elements to bind thus we have to create an element to use it. ui.monitor just watch the values.it's useful when we just want to watch the value change,and do something based on the data.    
 Refresh all bindings in an endless loop in the background increases the CPU load.    It is possible to learn from [ex4nicegui ](https://github.com/CrystalWindSnake/ex4nicegui)to trigger events only when data changes, which can reduce the CPU load.

**Comment by falkoschindler** [comment]
You can use our observable collections:

```py
from nicegui.observables import ObservableList

numbers = ObservableList([1, 2, 3], on_change=lambda: ui.notify('changed'))
ui.button('Change', on_click=lambda: numbers.append(4))
```

It has been developed for internal use and, thus, isn't well documented. But it should work nonetheless.
